cmd/mtbroker/filter/main.go

```diff
@@ -0,0 +1,168 @@
+)
+
+const (
+	defaultMetricsPort = 9092
+	component          = "broker_filter"
+)
+
+type envConfig struct {
+	Namespace string `envconfig:"NAMESPACE" required:"true"`
+	// TODO: change this environment variable to something like "PodGroupName".
+	PodName       string `envconfig:"POD_NAME" required:"true"`
+	ContainerName string `envconfig:"CONTAINER_NAME" required:"true"`
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := signals.NewContext()
+
+	// Report stats on Go memory usage every 30 seconds.
+	msp := metrics.NewMemStatsAll()
+	msp.Start(ctx, 30*time.Second)
+	if err := view.Register(msp.DefaultViews()...); err != nil {
+		log.Fatalf("Error exporting go memstats view: %v", err)
+	}
+
+	cfg, err := sharedmain.GetConfig(*masterURL, *kubeconfig)
+	if err != nil {
+		log.Fatal("Error building kubeconfig", err)
+	}
+
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Fatal("Failed to process env var", zap.Error(err))
+	}
+
+	ctx, _ = injection.Default.SetupInformers(ctx, cfg)
+	kubeClient := kubeclient.Get(ctx)
+
+	loggingConfigMapName := cmpresources.MakeCopyConfigMapName(namespaceresources.DefaultConfigMapPropagationName, logging.ConfigMapName())
+	metricsConfigMapName := cmpresources.MakeCopyConfigMapName(namespaceresources.DefaultConfigMapPropagationName, metrics.ConfigMapName())
+
+	//	loggingConfig, err := broker.GetLoggingConfig(ctx, env.Namespace, loggingConfigMapName)
+	loggingConfig, err := broker.GetLoggingConfig(ctx, "knative-eventing", loggingConfigMapName)
+	if err != nil {
+		log.Fatal("Error loading/parsing logging configuration:", err)
+	}
+	sl, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
+	logger := sl.Desugar()
+	defer flush(sl)
+
+	logger.Info("Starting the Broker Filter")
+
+	eventingClient := eventingv1alpha1.NewForConfigOrDie(cfg)
+	eventingFactory := eventinginformers.NewSharedInformerFactory(eventingClient,
+		controller.GetResyncPeriod(ctx))
+	triggerInformer := eventingFactory.Eventing().V1alpha1().Triggers()
+
+	// Watch the logging config map and dynamically update logging levels.
+	configMapWatcher := configmap.NewInformedWatcher(kubeClient, "knative-eventing")
+	// Watch the observability config map and dynamically update metrics exporter.
+	updateFunc, err := metrics.UpdateExporterFromConfigMapWithOpts(metrics.ExporterOptions{
+		Component:      component,
+		PrometheusPort: defaultMetricsPort,
+	}, sl)
+	if err != nil {
+		logger.Fatal("Failed to create metrics exporter update function", zap.Error(err))
+	}
+	configMapWatcher.Watch(metricsConfigMapName, updateFunc)
+	// TODO change the component name to broker once Stackdriver metrics are approved.
+	// Watch the observability config map and dynamically update request logs.
+	configMapWatcher.Watch(loggingConfigMapName, logging.UpdateLevelFromConfigMap(sl, atomicLevel, component))
+
+	bin := tracing.BrokerFilterName(tracing.BrokerFilterNameArgs{
+		Namespace:  env.Namespace,
```

Comment:
This is the only use of `Namespace`, is this MT yet?  Could it be `system.Namespace()` and drop that field?